File: backend/internal/domain/remix.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// RemixJob represents an async remix analysis job.
type RemixJob struct {
	ID           uuid.UUID
	WorkspaceID  uuid.UUID
	UserID       uuid.UUID
	Status       string // pending, ingesting, transcribing, scoring, generating, complete, failed
	InputURL     string
	Results      map[string]any
	ErrorMessage *string
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: frontend/packages/types/src/remix.ts
import type { UUID, Timestamp } from './common';

export type RemixJobStatus =
  | 'pending'
  | 'ingesting'
  | 'transcribing'
  | 'scoring'
  | 'generating'
  | 'complete'
  | 'failed';

export interface RemixJob {
  id: UUID;
  workspace_id: UUID;
  user_id: UUID;
  status: RemixJobStatus;
  input_url: string;
  results?: Record<string, unknown> | null;
  error_message?: string | null;
  created_at: Timestamp;
  updated_at: Timestamp;
}
